docs/DeveloperGuide.md

```diff
@@ -154,8 +154,84 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
-### \[Proposed\] Undo/redo feature
+### Add Gender
+
+The Add Gender feature allows users to add a gender field (Male / Female) to a person in the contact list. It is performed as a part 
+of `addCommand#execute()`.
+
+These operations are exposed in the `Model` interface as the method `Model#addPerson()`, which calls 
+`AddressBook#addPerson()` which calls `UniquePersonList#add` to add a new person in the person list 
+stored in AddressBook. 
+
+The following sequence diagram shows the methods calls related to add person operation: 
+
+![AddPersonSequenceDiagram](images/AddPersonSequenceDiagram.png)
```

Comment:
Also do you want to add the `AddPersonCommandParser` in between the `AddressBookParser` and `AddPersonCommand`? I feel that this interaction is important to be shown.